pkgs/tools/misc/ollama-webui/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, buildNpmPackage
+, nodePackages
+, fetchFromGitHub
+, runtimeShell
+}:
+
+# We just package the JS frontend part, not the Python reverse-proxy backend.
+# NixOS can provide any another reverse proxy such as nginx.
+buildNpmPackage rec {
+  pname = "ollama-webui";
+  # ollama-webui doesn't tag versions yet.
+  version = "0.0.0-unstable-2023-12-22";
+
+  src = fetchFromGitHub {
+    owner = "ollama-webui";
+    repo = "ollama-webui";
+    rev = "77c1a77fccb04337ff95440030cd051fd16c2cd8";
+    hash = "sha256-u7h2tpHgtQwYXornslY3CZjKjigqBK2mHmaiK1EoEgk=";
+  };
+  # dependencies are downloaded into a separate node_modules Nix package
+  npmDepsHash = "sha256-SI2dPn1SwbGwl8093VBtcDsA2eHSxr3UUC+ta68w2t8=";
+
+  # We have to bake in the default URL it will use for ollama webserver here,
+  # but it can be overriden in the UI later.
+  PUBLIC_API_BASE_URL = "http://localhost:11434/api";
+
+  # The path '/ollama/api' will be redirected to the specified backend URL
+  OLLAMA_API_BASE_URL = PUBLIC_API_BASE_URL;
+  # "npm run build" creates a static page in the "build" folder.
+  installPhase = ''
+    mkdir -p $out/lib
+    cp -R ./build/. $out/lib
+
+    mkdir -p $out/bin
```

Comment:
you don't need those last instructions.
I could be wrong, but if this is a standard node project, you just need to package the build directory
then in the service you can use node directly to just run the build directory
here are some reference of how we did this with lemmy
https://github.com/NixOS/nixpkgs/blob/aa5d1d03b1981f0968e3fdd0cd376c0a06fcaeef/pkgs/servers/web-apps/lemmy/ui.nix#L35
https://github.com/NixOS/nixpkgs/blob/aa5d1d03b1981f0968e3fdd0cd376c0a06fcaeef/nixos/modules/services/web-apps/lemmy.nix#L4

second point and this is entirely optional.
rather than making a complety separate service for ollama-webui, how about including it in the original ollama service.
I don't think it makes sense to have ollama-webui as a standalone for now.
(you can take inspiration in the lemmy module for what we did if you find stuff that you like).

Reply:
> second point and this is entirely optional.
> rather than making a complety separate service for ollama-webui, how about including it in the original ollama service.
> I don't think it makes sense to have ollama-webui as a standalone for now.
> (you can take inspiration in the lemmy module for what we did if you find stuff that you like).

Since `ollama-webui` is a community project and not officially associated with `ollama` (see the top of [their README.md](https://github.com/ollama-webui/ollama-webui)), I tend to think it does not make sense to have the `ollama-webui` service live under `ollama` as `ollama.ui` or `ollama.web-ui`.

There are a bunch of other frontends that people might want to use with the `ollama` service, such as `oterm`.